### DISCUSSION TITLE
TypeError: null is not an object (evaluating '_reactNative.NativeModules.ImagePickerManager.launchImageLibrary')

### DISCUSSION THREAD

**Discussion by ramijos** [react-native-image-picker/react-native-image-picker#1621] [Q&A] [answered]
Hello,
didi this API still working, please tell me and help solve this issue

**Comment by jadtay** [comment]
Also am having this problem when trying to implement the example code

**Comment by ramijos** [comment, marked as answer]
oufff!! i have  finally found a solution that has worked for me:
1- Edit the gradle file : "project_name\android\build.gradle":
				buildscript {
				    ext {
				        buildToolsVersion = "29.0.2"
				        minSdkVersion = 21
				        compileSdkVersion = 29
				        targetSdkVersion = 29
				    } 
#to avoid error: use a compatible library with a minsdk of at most 16

2- uninstall the lastest image-picker module and install the old one:
				- npm uninstall react-native-image-picker
				- npm install react-native-image-picker@2.3.4 --save
				- react-native link react-native-image-picker
				- npx react-native start --reset-cache
				
3- edit permissions:
			- ios: ios/project_name/Info.plist
                                <plist version="1.0">
                                    <dict>
                                         ...
                                          <key>NSPhotoLibraryUsageDescription</key>
                                          <string>$(PRODUCT_NAME) voudrait accéder à votre galerie photo</string>
                                          <key>NSCameraUsageDescription</key>
                                          <string>$(PRODUCT_NAME) voudrait utiliser votre caméra</string>
                                          <key>NSPhotoLibraryAddUsageDescription</key>
                                          <string>$(PRODUCT_NAME) voudrait enregistrer des photos dans votre galerie photo</string>
                                          <key>NSMicrophoneUsageDescription</key>
                                          <string>$(PRODUCT_NAME) voudrait accéder à votre micro (pour les vidéos)</string>
                                    </dict>
                                </plist>
                        - android: /android/app/src/main/AndroidManifest.xml

                             <uses-permission android:name="android.permission.CAMERA" />
                            <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>

4- compile app : npx react-native run-android
ps: use this doc for usage: https://github.com/react-native-image-picker/react-native-image-picker

**Comment by Johan-dutoit** [comment]
The latest version works just fine and is used in production daily.
